python/simpler/worker.py

```
@@ -2995,15 +3003,51 @@ def handle_control(sub_cmd: int) -> tuple[int, str]:
+            def _make_cleanup(_shm=shm, _pid=pid, _kind=kind):
+                def _cleanup_child():
+                    try:
+                        wpid, _status = os.waitpid(_pid, os.WNOHANG)
+                    except OSError:
+                        wpid = 0
+                    if wpid == 0:
+                        raise RuntimeError(f"child {_kind} pid {_pid} still alive; shm not freed")
+                    try:
+                        _shm.close()
+                    except Exception:
+                        pass
```

Ruff (0.16.0)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)